Source/RevitIFCTools/IFCEntityListWin.xaml.cs

```diff
@@ -161,7 +161,8 @@ private void button_Go_Click(object sender, RoutedEventArgs e)
 
             //Collect information on applicable Psets for Entity
             IDictionary<string, HashSet<string>> entPsetDict = new Dictionary<string, HashSet<string>>();
-            foreach(KeyValuePair<string, IList<VersionSpecificPropertyDef>> pdefEntry in procPdef.allPDefDict)
+            schemaEntities.PsetDefList.Add(DefineFXProperties());
```

Comment:
I have created an #if FORNAX_EXTENSION - #endif section below the foreach block. You should put this line of code that is specific to FORNAX into that section and remove the TODO line

Reply:
![image](https://user-images.githubusercontent.com/50314842/57440034-d44df980-7279-11e9-8509-1f045a88891c.png)

Sir after I added  schemaEntities.PsetDefList.Add(DefineFXProperties()); inside the #if FORNAX_EXTENSION - #endif section the IFCATTRIBUTE became duplicated in PropertySet Definition List.

Reply:
Yes probably because the #if - #endif block is inside a foreach block. You should just move the #if - #endif block to the place where you added schemaEntities.PsetDefList.Add(DefineFXProperties()); originally. 
My point is that I want to add any codes specific to FORNAX to be placed inside #if - #endif block so that they will only be used when the preprocessor is defined at compile time